src/main/java/xyz/nucleoid/bedwars/game/BwMapBuilder.java

```diff
@@ -43,12 +45,13 @@ private BwMap buildFromTemplate(MinecraftServer server, MapTemplate template) {
         BwMap map = new BwMap();
 
         MapTemplateMetadata metadata = template.getMetadata();
+        map.pools = new ItemGeneratorPools(config);
         metadata.getRegionBounds("diamond_spawn").forEach(map::addDiamondGenerator);
         metadata.getRegionBounds("emerald_spawn").forEach(map::addEmeraldGenerator);
 
         for (GameTeam team : this.config.teams()) {
             BwMap.TeamRegions regions = BwMap.TeamRegions.fromTemplate(team.key(), metadata);
-            map.addTeamRegions(team.key(), regions);
+            map.addTeamRegions(team.key(), regions, new ItemGeneratorPools(config));
```

Comment:
Ideally we don't need to reconstruct the pools object every time- can we reference the map directly here?